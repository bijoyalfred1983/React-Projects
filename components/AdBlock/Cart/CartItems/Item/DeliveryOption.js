/* @flow */
import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import type { Connector } from 'react-redux';
import styles from './DeliveryStyles.scss';
import type { Dispatch, ReduxState } from '../../../../types';
import * as itemQuantity from '../../../../actions/itemQuantity';

type Props = {
  data: Object,
  quantityData: Object,
  storePickup: () => void,
  quantityUpdateReadyStatus: string,
  commerceId: any,
  staticLabelData: Object
};

type State = {
  toggleAvailability: boolean
};
export class DeliveryOption extends PureComponent<Props, State> {
  constructor(props: any) {
    super(props);
    this.state = {
      toggleAvailability: this.props.data && this.props.data.storeItem
    };
  }

  componentWillReceiveProps(nextProps: Object) {
    if (nextProps.commerceId === this.props.data.commerceItemId) {
      if (nextProps.quantityData) {
        if (nextProps.quantityUpdateReadyStatus === 'STORECHANGE_SUCCESS') {
          this.setState({
            toggleAvailability: !this.state.toggleAvailability
          });
        }
      }
    }
  }

  toggleActiveRadio = value => {
    const self = this;
    // const { quantityUpdateReadyStatus } = self.props;
    return () => {
      const data = {
        type: value
      };
      self.props.storePickup(data, self.props.data.commerceItemId);
    };
  };

  newPart = () => {
    const self = this;
    self.setState({
      toggleAvailability: !self.state.toggleAvailability
    });
  };

  render() {
    const { data, staticLabelData } = this.props;
    const availabilityInfo = data.skuPricingAndAvailability;
    const storeClass = this.state.toggleAvailability
      ? styles.white_active
      : styles.grey_inactive;
    const storeClass1 = !this.state.toggleAvailability
      ? styles.white_active
      : styles.grey_inactive;

    return (
      <div className={styles.deliveryOption}>
        <form className={`row ${styles.display_flex}`}>
          <div className={`${styles.pickUpStore} ${storeClass}`}>
            <div className={styles.display_flex}>
              <input
                type="radio"
                id="radio-1"
                name="StoreBtn"
                value="ship"
                className={`${styles.storePickPD} ${styles.pickup} 
                ${
                  availabilityInfo.storePickupAvailable ? '' : styles.notAllowed
                }`}
                disabled={!availabilityInfo.storePickupAvailable}
                checked={this.state.toggleAvailability}
                onChange={this.toggleActiveRadio('store')}
              />
            </div>
            <div className={styles.storeLeft}>
              <div className={styles.store_pickUp} id="storePickUpID">
                {staticLabelData.label_cart_lineItem_StorePickUp
                  ? staticLabelData.label_cart_lineItem_StorePickUp
                  : 'label_cart_lineItem_StorePickUp'}
              </div>
              <div
                className={
                  availabilityInfo.storePickupStockLabel !== 'Out of Stock' &&
                  availabilityInfo.storePickupStockLabel !== 'Not Available'
                    ? styles.greenFont
                    : styles.redFont
                }
                id="storePickUpID"
              >
                {availabilityInfo.storePickupStockLabel}
              </div>
            </div>
          </div>
          <div className={`${styles.shipHomeStore} ${storeClass1}`}>
            <div>
              <input
                className={`with-gap ${styles.shipHomeCS} ${styles.pickup}
                  ${
                    availabilityInfo.shipToHomeAvailable
                      ? ''
                      : styles.notAllowed
                  }`}
                disabled={!availabilityInfo.shipToHomeAvailable}
                name="StoreBtn"
                type="radio"
                id="storeHomeID"
                value="store"
                checked={!this.state.toggleAvailability}
                onChange={this.toggleActiveRadio('online')}
              />
            </div>
            <div className={styles.storeLeft1}>
              <div className={styles.store_pickUp}>
                {staticLabelData.label_cart_lineItem_ShipToHomeHyphenated
                  ? staticLabelData.label_cart_lineItem_ShipToHomeHyphenated
                  : 'label_cart_lineItem_ShipToHomeHyphenated'}
              </div>
              <div
                className={
                  availabilityInfo.shipToHomeStockLabel !== 'Out of Stock' &&
                  availabilityInfo.shipToHomeStockLabel !== 'Not Available'
                    ? styles.greenFont
                    : styles.redFont
                }
              >
                {this.state.toggleAvailability ? (
                  <span>{availabilityInfo.shipToHomeStockLabel}</span>
                ) : null}

                {!this.state.toggleAvailability ? (
                  <span>
                    {data.estDeliveryTime.trim() === '' ? (
                      <span>{availabilityInfo.shipToHomeStockLabel}</span>
                    ) : (
                      <span>
                        <span className={styles.desktopOnly}>
                          {data.estDeliveryTime}
                        </span>
                        <span className={styles.mobileOnly}>
                          {data.estDeliveryTime.replace('. Delivery:', '')}
                        </span>
                      </span>
                    )}
                  </span>
                ) : null}
              </div>
            </div>
          </div>
        </form>
      </div>
    );
  }
}

/* istanbul ignore next */
const connector: Connector<{}, Props> = connect(
  (state: ReduxState) => ({
    quantityData: state.cartUpdate.storeData,
    quantityUpdateReadyStatus: state.cartUpdate.readyStatus,
    commerceId: state.cartUpdate.commerceId
  }),
  (dispatch: Dispatch) => ({
    storePickup: (value, id) => dispatch(itemQuantity.storePickup(value, id))
  })
);

export default connector(DeliveryOption);
