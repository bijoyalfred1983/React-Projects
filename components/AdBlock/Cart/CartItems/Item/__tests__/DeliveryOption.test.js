/* @flow */
import React from 'react';
import { createStore } from 'redux';
import { Provider, connect } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { DeliveryOption } from '../DeliveryOption';

describe('<DeliveryOption />', () => {
  let PROPS = {};
  let mockData = null;
  let ConnectedProduct;
  let mapStateToProps;
  let mapDispatchToProps;
  let staticLabelData;
  beforeEach(() => {
    staticLabelData = {
    "label_cart_lineItem_ShipToHomeHyphenated":"",
    "StorePickUpHyphenated":"",
    "label_cart_lineItem_StorePickUpHyphenated":""
  };
    mockData = {
      staticLabelData,
      data: {
        readyStatus: 'STORECHANGE_SUCCESS',
        storeItem: true,
        rawTotalPrice: 7.99,
        estDeliveryTime: 'Est. Delivery: Feb  14-15',
        displayName: 'Pilot Automotive/Antenna',
        availability: 'In Stock',
        commerceItemId: 'ci343000002',
        techNotes: null,
        productNotes: null,
        vehicleFit: false,
        imageURL:
          'https://contentinfo.autozone.com/znetcs/product-info/en/US/pil/CZ-241EC/image/1/',
        onSale: false,
        rebateAvailable: false,
        warrenty: null,
        eligibleForNextDay: false,
        amount: 7.99,
        '@class': 'com.autozone.diy.vo.AZCartCommerceItems',
        quantity: 1,
        salePrice: 0,
        dealAvailable: false,
        catalogRefId: '20372',
        application: null,
        unitAmount: 7.99,
        location: null,
        partNumber: 'CZ-241EC',
        quickNote: '425 Cold cranking Amps (530 cranking Amps)',
        currencyCode: 'USD',
        listPrice: 7.99,
        skuPricingAndAvailability: {
          lowestPrice: 0,
          corePriceAvailable: true,
          rebateUrl: null,
          vendorId: null,
          bopusStore: false,
          productLine: null,
          storePickupCSSClass: null,
          SRElegible: false,
          configurableSkuMessage: null,
          shipToHomeCSSClass: null,
          storePickupLabel: 'Store Pick Up',
          shipToHomeLabel: 'Ship To Home',
          skuId: '319462',
          highestPrice: 0,
          '@class': 'com.autozone.diy.commerce.pricing.AZPricingDealsInfo',
          onlineStockLabel: null,
          productId: null,
          rebatesAvaialble: false,
          shipToHomeStockLabel: 'In Stock',
          skuAddedTocart: false,
          storePickupAvailable: true,
          dealAvailable: false,
          skuUnavailable: false,
          storePickupStockLabel: 'Out of Stock',
          skuAvailabilityInfo: {
            unAvailableInStoreAndVdpStore: false,
            lowestPrice: null,
            unavailOnlineQoh: false,
            hubAvail: false,
            storeAvail: true,
            availableInOnline: false,
            webCorePrice: null,
            availableInVdpOnline: false,
            shippingAvailabilityMessage: {
              eligibleForNextDay: false,
              '@class':
                'com.autozone.diy.integration.shipping.ShippingAvailabilityMessage',
              inventoryMessage: null,
              holiday: false,
              cutOffTime: null,
              orderByMessage: null,
              calculatedDate: null,
              carrierHoliday: false,
              commerceItemId: null,
              qoHAvailable: false,
              displayDate: null,
              nextDayAvailableMessage: null,
              messageCutoffTime: null,
              messageDate: null
            },
            availableInStore: true,
            storeTotalPrice: '192.99',
            orderByMessage: null,
            showSearchAnotherStoreLink: false,
            storeRetailPrice: '174.99',
            unAvailableInOnlineAndVdpOnline: true,
            unavailQoh: true,
            eligibleForNextDay: false,
            item: null,
            '@class': 'com.autozone.diy.valueobject.SKUAvailabilityPriceInfo',
            webRetailPrice: null,
            webTotalPrice: null,
            availableForDirectShipping: false,
            exceptionItem: false,
            storeCorePrice: '18.0',
            availableInVdpStore: false,
            messageCutOffTime: null,
            nextDayAvailableMessage: null,
            skuid: '319462'
          },
          estimatedDeliveryMessage: 'Not Available',
          shipToHomeAvailable: false,
          dealUrl: false,
          configurableSku: false,
          retailPrice: 174.99,
          categoryId: null,
          corePrice: 18
        }
      }
    };

    PROPS = mockData;
  });
  test('renders correctly', () => {
    PROPS.staticLabelData = {
    "label_cart_lineItem_ShipToHomeHyphenated":"AAA",
    "StorePickUpHyphenated":"aaa",
    "label_cart_lineItem_StorePickUpHyphenated":"aaa"
  };
    const wrapper = mount(
      <StaticRouter context={{}}>
        <DeliveryOption {...PROPS} />
      </StaticRouter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test('check whether component will receive props is called correctly', () => {
    PROPS.staticLabelData = {"TEST":""};
    const wrapper = mount(
      <StaticRouter context={{}}>
        <DeliveryOption {...PROPS} />
      </StaticRouter>
    );
    var dat = {
        quantityData: "Message",
        quantityStatus: "STORECHANGE_SUCCESS"
    }; 
    wrapper.find(DeliveryOption).instance().componentWillReceiveProps(dat);
  });
});
