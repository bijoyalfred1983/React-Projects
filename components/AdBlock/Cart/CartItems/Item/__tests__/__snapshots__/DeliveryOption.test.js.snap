// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeliveryOption /> renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <MuiThemeProvider>
    <DeliveryOption
      data={
        Object {
          "@class": "com.autozone.diy.vo.AZCartCommerceItems",
          "amount": 7.99,
          "application": null,
          "availability": "In Stock",
          "catalogRefId": "20372",
          "commerceItemId": "ci343000002",
          "currencyCode": "USD",
          "dealAvailable": false,
          "displayName": "Pilot Automotive/Antenna",
          "eligibleForNextDay": false,
          "estDeliveryTime": "Est. Delivery: Feb  14-15",
          "imageURL": "https://contentinfo.autozone.com/znetcs/product-info/en/US/pil/CZ-241EC/image/1/",
          "listPrice": 7.99,
          "location": null,
          "onSale": false,
          "partNumber": "CZ-241EC",
          "productNotes": null,
          "quantity": 1,
          "quickNote": "425 Cold cranking Amps (530 cranking Amps)",
          "rawTotalPrice": 7.99,
          "rebateAvailable": false,
          "salePrice": 0,
          "skuPricingAndAvailability": Object {
            "@class": "com.autozone.diy.commerce.pricing.AZPricingDealsInfo",
            "SRElegible": false,
            "bopusStore": false,
            "categoryId": null,
            "configurableSku": false,
            "configurableSkuMessage": null,
            "corePrice": 18,
            "corePriceAvailable": true,
            "dealAvailable": false,
            "dealUrl": false,
            "estimatedDeliveryMessage": "Not Available",
            "highestPrice": 0,
            "lowestPrice": 0,
            "onlineStockLabel": null,
            "productId": null,
            "productLine": null,
            "rebateUrl": null,
            "rebatesAvaialble": false,
            "retailPrice": 174.99,
            "shipToHomeAvailable": false,
            "shipToHomeCSSClass": null,
            "shipToHomeLabel": "Ship To Home",
            "shipToHomeStockLabel": "In Stock",
            "skuAddedTocart": false,
            "skuAvailabilityInfo": Object {
              "@class": "com.autozone.diy.valueobject.SKUAvailabilityPriceInfo",
              "availableForDirectShipping": false,
              "availableInOnline": false,
              "availableInStore": true,
              "availableInVdpOnline": false,
              "availableInVdpStore": false,
              "eligibleForNextDay": false,
              "exceptionItem": false,
              "hubAvail": false,
              "item": null,
              "lowestPrice": null,
              "messageCutOffTime": null,
              "nextDayAvailableMessage": null,
              "orderByMessage": null,
              "shippingAvailabilityMessage": Object {
                "@class": "com.autozone.diy.integration.shipping.ShippingAvailabilityMessage",
                "calculatedDate": null,
                "carrierHoliday": false,
                "commerceItemId": null,
                "cutOffTime": null,
                "displayDate": null,
                "eligibleForNextDay": false,
                "holiday": false,
                "inventoryMessage": null,
                "messageCutoffTime": null,
                "messageDate": null,
                "nextDayAvailableMessage": null,
                "orderByMessage": null,
                "qoHAvailable": false,
              },
              "showSearchAnotherStoreLink": false,
              "skuid": "319462",
              "storeAvail": true,
              "storeCorePrice": "18.0",
              "storeRetailPrice": "174.99",
              "storeTotalPrice": "192.99",
              "unAvailableInOnlineAndVdpOnline": true,
              "unAvailableInStoreAndVdpStore": false,
              "unavailOnlineQoh": false,
              "unavailQoh": true,
              "webCorePrice": null,
              "webRetailPrice": null,
              "webTotalPrice": null,
            },
            "skuId": "319462",
            "skuUnavailable": false,
            "storePickupAvailable": true,
            "storePickupCSSClass": null,
            "storePickupLabel": "Store Pick Up",
            "storePickupStockLabel": "Out of Stock",
            "vendorId": null,
          },
          "storeItem": true,
          "techNotes": null,
          "unitAmount": 7.99,
          "vehicleFit": false,
          "warrenty": null,
        }
      }
    >
      <div
        className="deliveryOption"
      >
        <form
          className="row display_flex"
        >
          <div
            className="pickUpStore white_active"
          >
            <div
              className="display_flex"
            >
              <input
                checked={true}
                className="test5 pickup
                "
                disabled={false}
                id="radio-1"
                name="myRadio"
                onChange={[Function]}
                type="radio"
                value="ship"
              />
            </div>
            <div
              className="storeLeft"
            >
              <div
                className="store_pickUp"
                id="test5"
              >
                STORE-PICK-UP
              </div>
              <div
                className="redFont"
                id="test5"
              >
                Out of Stock
              </div>
            </div>
          </div>
          <div
            className="shipHomeStore grey_inactive"
          >
            <div>
              <input
                checked={false}
                className="with-gap test51 pickup
                  notAllowed"
                disabled={true}
                id="test6"
                name="myRadio"
                onChange={[Function]}
                type="radio"
                value="store"
              />
            </div>
            <div
              className="storeLeft1"
            >
              <div
                className="store_pickUp"
              >
                SHIP-TO-HOME
              </div>
              <div
                className="greenFont"
              >
                In Stock
              </div>
            </div>
          </div>
        </form>
      </div>
    </DeliveryOption>
  </MuiThemeProvider>
</Provider>
`;
